src/Chocolatey.PowerShell/Shared/ChocolateyCmdlet.cs

```diff
@@ -27,6 +29,28 @@ namespace Chocolatey.PowerShell.Shared
     /// </summary>
     public abstract class ChocolateyCmdlet : PSCmdlet
     {
+        // Place deprecated command names and their corresponding replacement in this dictionary to have those commands
+        // warn users about the deprecation when they are called by those names.
+        private readonly Dictionary<string, string> _deprecatedCommandNames = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            // Use the following format to provide a deprecation notice. If the new command name is an empty string,
+            // the warning will inform the user it is to be removed instead of renamed.
+            //
+            // { "Deprecated-CommandName", "New-CommandName" },

```

Comment:
In the linked patch file in the PR description, there is an additional line to add to this _deprecatedCommandNames dictionary.  What is the reason that this is not included in this PR?  Just so that we can see it in action, before and after?

Reply:
Yep, just to demonstrate that it works, as we don't have any fully deprecated command names just yet.

Reply:
We could arguably mark anything we rename immediately as a deprecated name and just have it be aliased until next major version, but wasn't sure if we wanted to do that yet.